netbox_diode_plugin/views.py

```diff
@@ -109,3 +120,234 @@ def post(self, request, *args, **kwargs):
         kwargs["pk"] = settings.pk
 
         return super().post(request, *args, **kwargs)
+
+
+class GetReturnURLMixin:
+    """Get return URL mixin."""
+
+    def get_return_url(self, request):
+        """Get return URL."""
+        # First, see if `return_url` was specified as a query parameter or form data. Use this URL only if it's
+        # considered safe.
+        return_url = request.GET.get("return_url") or request.POST.get("return_url")
+        if return_url and url_has_allowed_host_and_scheme(
+            return_url, allowed_hosts=None
+        ):
+            return return_url
+
+        return None
+
+
+class BaseDiodeView(View):
+    """Base diode view."""
+
+    def check_authentication(self, request):
+        """Check authentication."""
+        if not request.user.is_authenticated or not request.user.is_staff:
+            next_url = request.path
+            if not url_has_allowed_host_and_scheme(next_url, allowed_hosts=None):
+                next_url = "/"
+
+            safe_redirect_url = f"{netbox_settings.LOGIN_URL}?next={next_url}"
+            return redirect(safe_redirect_url)
```

Comment:
We already have `redirect_to_login()` function in this view (that was also fixed due to similar code scanning issues  in the past), can we try this:

```suggestion
        if not request.user.is_authenticated or not request.user.is_staff:
            return redirect_to_login(request)
```